File: Groove.CSFE/Groove.CSFE.Core/Entities/WarehouseAssignmentModel.cs
namespace Groove.CSFE.Core.Entities
{
    public class WarehouseAssignmentModel : Entity
    {
        public long WarehouseLocationId { get; set; }
        public long OrganizationId { get; set; }
        public string ContactPerson { get; set; }
        public string ContactPhone { get; set; }
        public string ContactEmail { get; set; }
        public OrganizationModel Organization { get; set; }
        public WarehouseLocationModel WarehouseLocation { get; set; }
    }
}

File: Groove.SP/groove-sp-web/src/app/core/models/warehouse-assignment/warehouse-assignment.model.ts
import { WarehouseLocationModel } from "../warehouse-locations/warehouse-location.model";

export interface WarehouseAssignmentModel {
    warehouseLocationId: number;
    organizationId: number;
    contactPerson: string;
    contactPhone: string;
    contactEmail: string;
    warehouseLocation: WarehouseLocationModel;

    // using on UI

    /**to mark this as being adding */
    isAddLine: boolean;
}